### PR TITLE
blog: introduce CSI support for ephemeral inline volumes

### PR DESCRIPTION
**Pull request by pohly**
This was alpha in Kubernetes 1.15 and became beta in 1.16. Several CSI
drivers already support it.

/cc @kfox1111 @msau42

### CODE REVIEW

File: content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+*ephemeral inline* volumes. The [CSIInlineVolume feature
+gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+had to be set to enable it in 1.15 because support was still in alpha
+state. In 1.16, the feature reached beta state, which typically means
+that it is enabled in clusters by default.
+
+CSI drivers have to be adapted to support this because although an
+existing CSI gRPC call (`NodePublishVolume`) is used, the way how it
+is used is different and not covered by the CSI spec: for ephemeral
+volumes, only that call is invoked by `kubelet` and all other calls
```
Comment (kfox1111) [inline content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md]: Minor nit. NodeUnpublishVolume is also called.

File: content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+(PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
+objects. Originally, volumes that are backed by a Container Storage Interface
+(CSI) driver could only be used via this PVC/PV mechanism.
+
+But there are also use cases for data volumes whose content and
+lifecycle is tied to a pod. For example, a driver might populate a
+volume with dynamically created secrets that are specific to the
+application running in the pod. Such volumes need to be created
+together with a pod and can be deleted as part of pod termination
+(*ephemeral*). They get defined as part of the pod spec (*inline*).
```
Comment (kfox1111) [inline content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md]: Might be good to give an example, like: https://github.com/kubernetes-csi/csi-driver-image-populator#example-usage-in-kubernetes

File: content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md (hunk 1 above)
Comment (pohly) [reply content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md]: Indeed, will fix that.

File: content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md (hunk 2 above)
Comment (pohly) [reply content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md]: I was hoping that readers will patient enough to continue reading and will then find real examples in the currently missing sections for PMEM-CSI, image populator and cert manager.

Comment (kfox1111) [reply content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md]: That works too.

**Comment by pohly** [comment]
I finished the blog post by fleshing out the example sections and adding a conclusion ("next steps").

/assign @kfox1111

@kfox1111: can you have another look?

File: content/en/blog/_posts/2020-01-21-csi-ephemeral-inline-volumes.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+```sh
+kubectl exec nginx -- cat /usr/share/nginx/html/test
+```
+
+That `test` file just contains a single word:
+```
+testing
+```
+
```
Comment (kfox1111) [inline content/en/blog/_posts/2020-01-21-csi-ephemeral-inline-volumes.md]: Maybe one more little bit:

Data containers can be built with Dockerfiles such as:
```yaml
FROM scratch
COPY index.html /index.html
```

Comment (pohly) [reply content/en/blog/_posts/2020-01-21-csi-ephemeral-inline-volumes.md]: Added.

**Comment by mrbobbytables** [comment]
/lgtm 
from a technical perspective things look good 👍
@kbarnard10 do you want to do the final sign off?

**Pull request merged**